Use consistent whitespace in template placeholders

fhemberger opened the pull request:
Sorry, I know this a trivial PR, but it would be nice to have consistent use of whitespace inside all charts. 😉

fhemberger committed:
Use consistent whitespace in template placeholders

unguiculus commented:
Thanks a lot. This is very much appreciated. But I'm afraid most of this, if not all, has already been done in #1062 which has just been merged today.

fhemberger commented:
@unguiculus Hmm, the issue you mentioned is about Elasticsearch. 🤔

unguiculus commented:
Ouch, sorry. I guess I got confused by the merge conflicts. Here's an open PR for Artifactory which should get merged soon: #1314.

fhemberger commented:
@unguiculus I updated the PR to resolve the merge conflicts. My PR doesn't just cover Artifactory, it fixes the template placeholder whitespace in *all* charts.

unguiculus commented:
Wow, I guess I did not look far enough. Did you use some script for this? 

Usually we want separate PRs per chart. Do you think you can split this up?

@kubernetes/charts-maintainers How should we go about this? I think this is great.

fhemberger commented:
Just a simple Regex search & replace in Sublime Text:

`{{([^ -])` -> `{{ $1`
`([^ -])}}` -> `$1 }}`
(search for all double curly braces that are **not** followed by a space or a dash and insert a single space between them and the char found)

Splitting it up would have ended up in 38 separate PRs just adding whitespace, so I left it in one single PR instead. 😉

https://github.com/kubernetes/charts/pull/1437/files?w=1 shows that it's really only whitespace that has changed.

unguiculus commented:
Maybe we can take this as a whole. Discussing it on Slack.

unguiculus commented on incubator/etcd/templates/etcd-statefulset.yaml:
```diff
@@ -129,7 +129,7 @@ spec:
 
             # etcd-SET_ID
-            SET_ID=${HOSTNAME:5:${#HOSTNAME}}
+            SET_ID=${HOSTNAME:5:${#HOSTNAME }}
```
This is shell not Go templating. We probably don't want to change that.

fhemberger replied:
Fixed.

unguiculus reviewed (changes requested):
I just found one place where the replace should not happen.

mgoodness commented:
/test pull-charts-e2e

fhemberger committed:
[incubator/tensorflow-inception] Fix template helper

fhemberger commented:
Rebased master, fixed tensorflow helper.

/retest

unguiculus commented:
/retest

fhemberger commented:
@unguiculus Sorry, not sure what's wrong with the Jenkins built, I don't think it's actually related to my changes.

unguiculus commented:
Yeah, there seems to be some test flakiness. We are going to merge this as a whole. Anything else does not make sense.

unguiculus commented:
Would you mind covering `NOTES.txt` files as well? I understand that those will not be as easy because they often contain shell magic. But we can just try and get this in soon and follow up with that in a separate PR.

fhemberger commented:
Sure, I can do that as well, but I'd prefer a separate PR to minimize impact.

fhemberger committed:
Fix whitespace in NOTES.txt

Also make sure paramater order is always

`kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} --template "…"`

fhemberger commented:
@unguiculus Fixing NOTES.txt wasn't so bad, so I added it here. Also fixed parameter order on three occasions, so the command is always identical. Should I also adjust `APP_HOST` vs. `SERVICE_IP`? (I'd prefer the latter, it is more precise.)

unguiculus commented:
I'm going to merge this as is. You may create a separate PR as follow-up. I'd prefer SERVICE_IP as well.

fhemberger commented:
Awesome, thanks!

rk295 commented:
Hi, this commit breaks (at least) the grafana chart. It fails a lint with the default values. 

```==> Linting grafana/
[ERROR] templates/: parse error in "grafana/templates/job.yaml": template: grafana/templates/job.yaml:41: unexpected "}" in operand

Error: 1 chart(s) linted, 1 chart(s) failed
```

rk295 commented:
As a follow up, would it be worth having CI lint each of the charts in branches before merging?

fhemberger commented:
@rk295 Sorry, didn't notice the change in the Job.
Fixed in #1458.
